gateway/proxy_muxer.go

```diff
@@ -404,15 +404,28 @@ func target(listenAddress string, listenPort int) string {
 	return fmt.Sprintf("%s:%d", listenAddress, listenPort)
 }
 
-func (m *proxyMux) generateListener(listenPort int, protocol string) (l net.Listener, err error) {
-	listenAddress := config.Global().ListenAddress
-	disabled := config.Global().DisabledPorts
-	for _, d := range disabled {
-		if d.Protocol == protocol && d.Port == listenPort {
-			return nil, fmt.Errorf("%s:%d trying to open disabled port", protocol, listenPort)
+func CheckPortWhiteList(cfg config.Config, listenPort int, protocol string) error {
+	if protocol == "" || protocol == "http" || protocol == "https" {
```

Comment:
Instead of this logic, lets just set PortWhiteList vars on the gateway start. Like (pseudocode):
```
protocol = "http"
if cfg.SSLEnabled {
protocol = "https"
}

   ports := cfg.PortWhiteList[protocol].Ports
   ports := append(ports, cfg.ListenPort)
   cfg.PortWhiteList[protocol].Ports = ports
```